pipeline.go

```diff
@@ -345,59 +345,96 @@ func (p *PipelineDefinition) Pipelines() (*Pipelines, error) {
 	return p.pipelines, nil
 }
 
-func runParallelBuildImage(runner Runner, step Step, pull bool, durations *sync.Map, wg *sync.WaitGroup, start chan struct{}, abort chan struct{}) {
-	defer wg.Done()
-	<-start
+type runConfig struct {
+	usePreconditions bool
+	selection        func(step Step) bool
+	pre              func(runner Runner, step Step) error
+	run              func(runner Runner, step Step) func() error
+	post             func(runner Runner, step Step) error
+}
 
-	select {
-	case <-abort:
+func runCommandParallel(config runConfig, runner Runner, step Step, durations *sync.Map, wg *sync.WaitGroup, preconditions []chan struct{}, done chan struct{}, abort chan error) {
+	defer wg.Done()
+	defer close(done)
+	for i, c := range preconditions {
+		if Verbose {
+			pipelineLogger.Printf("%s waiting for %d preconditions", step.ColoredContainerName(), len(preconditions)-i)
```

Comment:
Shouldn't this be the singluar in the Printf as it prints the number of the current precondition?

Reply:
```
x waiting for 1 precondition
x waiting for 1 precondition(s)
x waiting for 1 preconditions

x waiting for 4 precondition
x waiting for 4 precondition(s)
x waiting for 4 preconditions
```
I'd suggest adding the `()`s to the message or did you mean a different place?